spark/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala

```diff
@@ -542,6 +542,52 @@ case class AlterTableChangeColumnDeltaCommand(
   }
 }
 
+/**
+ * A command to replace columns for a Delta table, support changing the comment of a column,
+ * reordering columns, and loosening nullabilities.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   ALTER TABLE table_identifier REPLACE COLUMNS (col_spec[, col_spec ...]);
+ * }}}
+ */
+case class AlterTableReplaceColumnsDeltaCommand(
+    table: DeltaTableV2,
+    columns: Seq[StructField])
+  extends LeafRunnableCommand with AlterDeltaTableCommand with IgnoreCachedData {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val deltaLog = table.deltaLog
+    recordDeltaOperation(deltaLog, "delta.ddl.alter.replaceColumns") {
```

Comment:
single-use val considered annoying if it doesn't help with line breaks:
```suggestion
    recordDeltaOperation(table.deltaLog, "delta.ddl.alter.replaceColumns") {
```